import-scripts/updated_filename_for_datatype_test.py

```diff
@@ -0,0 +1,534 @@
+meta_cna_foundation.txt
+meta_cna_rae.txt
+meta_cna_consensus.txt
+meta_cna.txt
+meta_linear_cna.txt
+meta_log2_cna.txt
+meta_armlevel_cna.txt
+meta_drug_treatment_auc.txt
+meta_drug_treatment_ic50.txt
+meta_drug_treatment_zscore.txt
+meta_mutations_uniprot_canonical_transcripts.txt
+meta_mutations_uniprot_canonical_transcripts.txt
+meta_mutations_uniprot_canonical_transcripts.txt
+meta_gene_matrix.txt
+meta_gene_panel_matrix.txt
+"""
+
+def filename_maps():
+    datatype_lines = datatype_column_content().splitlines()
+    staging_file_lines = staging_filename_column_content().splitlines()
+    meta_file_lines = meta_filename_column_content().splitlines()
+    if len(datatype_lines) != len(staging_file_lines):
+        print("Error : datatype_column_content linecount (%d) does not match staging_filename_column_content linecount (%d)" % (len(datatype_lines), len(staging_file_lines)))
+        sys.exit(1);
+    if len(datatype_lines) != len(meta_file_lines):
+        print("Error : datatype_column_content linecount (%d) does not match meta_filename_column_content linecount (%d)" % (len(datatype_lines), len(meta_file_lines)))
+        sys.exit(1);
+    filename2datatype = {}
+    datatype_seen = set([])
+    filename_is_first_listed = set([])
+    for index in range(1,len(datatype_lines)):
+        datatype = datatype_lines[index].strip();
+        if len(datatype) == 0:
+            continue
+        staging_filename = staging_file_lines[index].strip();
+        meta_filename = meta_file_lines[index].strip();
+        if not datatype in datatype_seen:
+            filename_is_first_listed.add(staging_filename)
+            filename_is_first_listed.add(meta_filename)
+        datatype_seen.add(datatype)
+        filename2datatype[staging_filename] = datatype
+        filename2datatype[meta_filename] = datatype
+    return filename2datatype, filename_is_first_listed
+
+def is_normal_sample_filename(filename):
+    normal_sample_suffixes = ["_normals.txt", "data_mrna_seq_v2_rsem_normal_samples.txt", "data_mrna_seq_v2_rsem_normal_samples_zscores_ref_normal_samples.txt"]
+    for suffix in normal_sample_suffixes:
+        if (filename.endswith(suffix)):
+            return True;
+    return False;
+
+def directory_uses_updated_filenames(path):
+    new_filenames_present = False
+    if os.path.isdir(path):
+        path = path + os.path.sep
+    basedir = os.path.dirname(path)
+    filename2datatype, filename_is_first_listed = filename_maps()
+    files = os.listdir(path)
+    for file in files:
+        filepath = os.path.join(basedir, file)
+        if not os.path.isfile(filepath):
+            # ignore subdirectories
+            continue
+        if not file in filename2datatype:
+            # filename/pattern was not found
+            continue
+        if is_normal_sample_filename(file):
+            # normal sample files are skipped in importer
+            continue
+        if not file in filename_is_first_listed:
```

Comment:
Think it's dangerous to have order in the google spreadsheet indicate "new vs old"

Reply:
agreed ... the familiar old file names will be hardcoded instead.